server/clientRoutes.js

```diff
@@ -0,0 +1,100 @@
+const express = require('express');
+const assert = require('bsert');
+const Config = require('bcfg');
+const logger = require('./logger');
+
+function clientsRouter(clients) {
+  const router = express.Router({ mergeParams: true });
+  let token, id, config, reqClients;
+
+  // return object with
+  // relevant info for each client
+  // keyed to the client id
+  const clientInfo = {};
+  clients.forEach(
+    client =>
+      (clientInfo[client.config.str('id')] = {
+        chain: client.config.str('chain', 'bitcoin')
```

Comment:
Would it be a hard to find bug if the user forgets to set the `chain` and it automatically sets as `bitcoin` when they didn't intend that? Should `chain` be a required config option?

Reply:
Yeah, I think that's a good observation. My only concern is that this then makes this non-backwards compatible as you would have to update any existing configs and I was trying as much as possible as I chipped away at this new feature to maintain existing functionality. I think the concern is valid though so if you think it's worth it I can make it required. Alternatively we could log a warning that says `chain` was not set and will set it to 'bitcoin' as a default.

Reply:
I like that idea, a logger warning will make it easy to debug